homeassistant/components/ambient_network/sensor.py

```diff
@@ -0,0 +1,305 @@
+"""Support for Ambient Weather Network sensors."""
+from __future__ import annotations
+
+from datetime import datetime
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+    CONCENTRATION_PARTS_PER_MILLION,
+    CONF_MAC,
+    DEGREE,
+    PERCENTAGE,
+    UnitOfIrradiance,
+    UnitOfLength,
+    UnitOfPrecipitationDepth,
+    UnitOfPressure,
+    UnitOfSpeed,
+    UnitOfTemperature,
+    UnitOfVolumetricFlux,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util import dt as dt_util
+
+from .const import DOMAIN
+from .coordinator import AmbientNetworkDataUpdateCoordinator
+from .entity import AmbientNetworkEntity
+
+TYPE_AQI_PM25 = "aqi_pm25"
+TYPE_AQI_PM25_24H = "aqi_pm25_24h"
+TYPE_BAROMABSIN = "baromabsin"
+TYPE_BAROMRELIN = "baromrelin"
+TYPE_CO2 = "co2"
+TYPE_DAILYRAININ = "dailyrainin"
+TYPE_DEWPOINT = "dewPoint"
+TYPE_EVENTRAININ = "eventrainin"
+TYPE_FEELSLIKE = "feelsLike"
+TYPE_HOURLYRAININ = "hourlyrainin"
+TYPE_HUMIDITY = "humidity"
+TYPE_LASTRAIN = "lastRain"
+TYPE_LIGHTNING_DISTANCE = "lightning_distance"
+TYPE_LIGHTNING_PER_DAY = "lightning_day"
+TYPE_LIGHTNING_PER_HOUR = "lightning_hour"
+TYPE_MAXDAILYGUST = "maxdailygust"
+TYPE_MONTHLYRAININ = "monthlyrainin"
+TYPE_PM25 = "pm25"
+TYPE_PM25_24H = "pm25_24h"
+TYPE_SOLARRADIATION = "solarradiation"
+TYPE_TEMPF = "tempf"
+TYPE_UV = "uv"
+TYPE_WEEKLYRAININ = "weeklyrainin"
+TYPE_WINDDIR = "winddir"
+TYPE_WINDGUSTMPH = "windgustmph"
+TYPE_WINDSPEEDMPH = "windspeedmph"
+TYPE_YEARLYRAININ = "yearlyrainin"
+
+
+SENSOR_DESCRIPTIONS = (
```

Comment:
It's a lot of sensor. Does all make sense directly or should some maybe be disabled on setting up the integration first time?

Reply:
Are you suggesting that we give the user the option to individually pick the sensors they want in the onboarding flow? How would a user add or remove a sensor later?

Reply:
You can disable entities by default. Users can then enable them when they want to.

entity_registry_enabled_default or something like that

Reply:
I see. I have disabled some sensors by default.